shell/platform/darwin/common/framework/Source/FlutterChannelsTest.m

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#if !__has_feature(objc_arc)
+#error ARC must be enabled!
+#endif
+
+#import <OCMock/OCMock.h>
+#import <XCTest/XCTest.h>
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterChannels.h"
+
+@interface MockBinaryMessenger : NSObject <FlutterBinaryMessenger>
+@property(nonatomic, copy) NSString* channel;
+@property(nonatomic, strong) NSData* message;
+@property(nonatomic, strong) NSMutableDictionary<NSString*, FlutterBinaryMessageHandler>* handlers;
+@end
+@implementation MockBinaryMessenger
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _handlers = [[NSMutableDictionary<NSString*, FlutterBinaryMessageHandler> alloc] init];
+  }
+  return self;
+}
+
+- (void)sendOnChannel:(NSString*)channel message:(NSData* _Nullable)message {
+  [self sendOnChannel:channel message:message binaryReply:nil];
+}
+
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
+  self.channel = channel;
+  self.message = message;
+}
+
+- (void)setMessageHandlerOnChannel:(NSString*)channel
+              binaryMessageHandler:(FlutterBinaryMessageHandler _Nullable)handler {
+  [self.handlers setObject:handler forKey:channel];
+}
+@end
+
+@interface FlutterChannelsTest : XCTestCase
+@end
+
+@implementation FlutterChannelsTest
+
+- (void)testMethodInvoke {
+  NSString* channelName = @"foo";
+  id binaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+  NSData* encodedMethodCall = [@"hey" dataUsingEncoding:NSUTF8StringEncoding];
+  OCMStub([codec encodeMethodCall:[OCMArg any]]).andReturn(encodedMethodCall);
```

Comment:
You can use `OCMOCK_ANY` instead of `[OCMArg any]`, but their docs have both.  Doesn't matter, just a fun fact.